### DISCUSSION TITLE
what is the steps to draw roads between two points ?

### DISCUSSION THREAD

**Discussion by jesussmile** [liodali/osm_flutter#106] [Q&A]
Hi there hope you are doing well,
I can't reproduce the steps to draw roads from the source code you have on GitHub. As I open the map and select two points, I don't immediately get a road, I see you have added a middle point to pass a route, is that the default behavior?  Also, I can't always drop a marker, I am probably doing it wrong, Kindly could you tell us how do you want the users to draw a road between two points?

**Comment by liodali** [comment]
can explain more about drop marker ?
those marker are showed when you drawRoad or that you picked !

**Comment by liodali** [comment]
```dart 
 RoadInfo roadInfo = await controller.drawRoad( 
   GeoPoint(latitude: 47.35387, longitude: 8.43609),
   GeoPoint(latitude: 47.4371, longitude: 8.6136),
   roadOption: RoadOption(
       roadWidth: 10.0,
       roadColor: Colors.blue,
       showMarkerOfPOI: false
   ),
);
showMarkerOfPoi here is if you already have marker and you don't want to show default marker that you can customize when you initialize map


```

**Reply by liodali** [reply]
you don't get the road maybe because the request take long time but i will check in my side 
maybe i need to optimize my code

**Reply by jesussmile** [reply]
I am not sure what is happening, I cannot pick two points to draw a road.  Please help me what am I doing wrong here? To better provide you the issue I have posted a youtube video. At first, I thought it was due to some of the .png files you were using from Github as pointers, changed it still no luck.  All I have done is run the app and trying to draw a road on the map. It works sometimes but I can't duplicate the steps

[https://www.youtube.com/watch?v=tSo_MIWSbiM](url)

**Reply by liodali** [reply]
the link not working can you share some code

**Reply by jesussmile** [reply]
if you copy the link and paste it into the browser you will get the video.

**Reply by liodali** [reply]
did you use await selectPosition !

**Reply by liodali** [reply]
what's the version that you are currently use is last one ?

**Reply by jesussmile** [reply]
I tried it with 0.8.0 alpha 0 ,  0.7.8 -2 , and 0.8.0 alpha 1 The touch gesture is intermittent. sometimes it works some times it doesn't. I have just begun to write codes to implement road but first I wanted to check how you are doing it.

**Reply by liodali** [reply]
for alpha version wait just for 2 week, i will make it more stable 
but for 0.7.8+2, i will check if is their any problem with the selection 
but already started working in add callback for tapping in map and i have some issues but it will shipped next week

**Reply by liodali** [reply]
but i will check 0.7.8+2 mayebe when i add it the  copyright widget 
i miss up with selection

**Reply by jesussmile** [reply]
Apologies, it's an issue on my side of incorrect implementation.

**Comment by jesussmile** [comment]
Oh okay! That's perfectly fine.
